src/Lean/Server/InfoUtils.lean

```diff
@@ -171,6 +171,16 @@ partial def InfoTree.hoverableInfoAt? (t : InfoTree) (hoverPos : String.Pos) (in
       if ti.expr.isSyntheticSorry then
         return none
   return res?
+where containsHoverPos info :=
+  if info.contains hoverPos includeStop then
+    info.range?
+  -- `(· + ·)` expands to a `+` node that has a synthetic range because its first and last token are synthetic.
```

Comment:
Is this an argument to store parser ranges for nodes as well?

The heuristic here seems a bit fragile to be honest.

Reply:
Mmh, you mean we should store the entire span as original info in the `+` node? That could work, yeah.

Reply:
This would however be a pretty fundamental change for a pretty rare use case: modifying a node's children without recreating the node as part of a quotation. I think in most cases we're trying to move away from this pattern. `expandCDot?` really is the only case that comes to mind where we do a generic traversal over the syntax tree.

Reply:
I agree that it's a bigger change.  It also has the advantage that it is consistent with how we store synthetic positions from macro expansions (which are put on the nodes).

> generic traversal over the syntax tree

We also need to do this for `(← ..)` and for `$(f a)`, and really for any "two-part" syntax.

> I think in most cases we're trying to move away from this pattern.

From what I can tell, there are quite a few places where we explicitly use this pattern to preserve separators, atom source infos, syntax variants (unicode/non-unicode, semicolon/newline), etc.

Reply:
> We also need to do this for (← ..) and for $(f a), and really for any "two-part" syntax.

Ah, this is fun: `x + (← y)` is fine since we do not touch the parentheses. And infos around antiquotations shouldn't matter in practice anyway, no?

> From what I can tell, there are quite a few places where we explicitly use this pattern to preserve separators, atom source infos, syntax variants (unicode/non-unicode, semicolon/newline), etc.

I'm not directly aware of that, it might be incidental.